oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/geometry/OSHDBGeometryBuilder.java

```diff
@@ -485,61 +487,57 @@ private static Polygon constructMultipolygonPart(
 
   // helper that joins adjacent osm ways into linear rings
   private static List<LinkedList<OSMNode>> buildRings(Stream<Stream<OSMNode>> lines) {
-    // make a (mutable) copy of the polygons array
+    // make a (mutable) copy of the lines list
     List<LinkedList<OSMNode>> ways = lines
         .map(line -> line.collect(Collectors.toCollection(LinkedList::new)))
         .filter(nodesList -> !nodesList.isEmpty())
         .collect(Collectors.toCollection(LinkedList::new));
 
     List<LinkedList<OSMNode>> joined = new LinkedList<>();
+    // iterate until there are no more ways left to process
     while (!ways.isEmpty()) {
```

Comment:
Is it theoretically possible, that `ways` never gets empty and stays in an infinite loop? I couldn't find any `break` or `return` in this loop. Normally, this should never be an issue, because of the next line.

Reply:
yeah, on each iteration, at least one element gets removed from the list (line 499). so the loop will definitely eventually be done. I tried to write this a bit differently to make it more obvious (e.g. by use of a `for (…; …; …)` loop, but in this particular situation it wasn't very practical/helpful.